src/reports/package/accessibilityInsightsReport.d.ts

```diff
@@ -47,16 +47,50 @@ declare namespace AccessibilityInsightsReport {
     };
 
     export type SummaryReportParameters = {
-        serviceName: string;
-        axeVersion: string;
-        userAgent: string;
-        crawlDetails: CrawlSummaryDetails;
-        results: SummaryScanResults;
+        serviceName: string,
+        axeVersion: string,
+        userAgent: string,
+        crawlDetails: CrawlSummaryDetails,
+        results: SummaryScanResults,
     };
 
+    export type AxeRuleData = {
+        ruleId: string,
+        tags: string[],
+        description: string,
+        ruleUrl: string
+    }
+
+    export type InstanceFailureData = {
+        urls: string[],
+        elementSelector: string,
+        snippet: string,
+        fix: string,
+    }
+
+    export type RuleFailureData = {
+        rule: AxeRuleData,
+        failures: InstanceFailureData[],
+    }
+
+    export type CombinedReportResults = {
```

Comment:
How this structure will handle two king of grouping for report UI - selector and rule? Does report code will do filtering and sorting from this combined data list?

Reply:
Since selector groupings are a stretch goal for this feature, we'll start with only accepting groupings by rule and add the selector groupings when we reach that point. I edited to make it easier to add more result groupings in the future.